File: 125-valid-palindrome/valid-palindrome.cpp
class Solution {
public:
    bool check(string s)
    {
        int n = s.length();
        for(int i = 0;i<n/2;i++)
        {
            if(s[i] != s[n-i-1 ])
            {
                return false;
            }
        }
        return true;
    }
    bool isPalindrome(string s) {
        int n = s.length();
        string st;
        for(auto &ch:s)
        {
          if((ch>=97 && ch<=122)||(ch>=48&&ch<=57))
              st+=ch;
          else if (ch>=65 && ch <=90)
              st+=(ch+32);
        }
        return check(st);
    }
};

File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        r = ""
        for c in s:
            if c.isdigit():
                r += c
            if c.isalpha():
                r += c.lower()
        r1 = ""
        for i in range(len(r)-1, -1, -1):
            r1 += r[i] 
        return r == r1
        